### PR TITLE
Modifier tap hold in keyboard toml

### PR DESCRIPTION
**Pull request by hariseldon78**
This pull request closes issue #221.

I have been able to test the MTH(A,LShift) and it works correctly, but for some reason the TH(X,Y) is not working yet: has it been tested before? I see that Via does not have a binding for the TapHold.

I used all caps for MTH and TH for consistency with the other Layer actions, and i used the ordering <key>, <modifiers> for consistency with WM and LM.

### CODE REVIEW

**Comment by hariseldon78** [comment]
i also fixed Lgui and Rgui to LGui to RGui, let me know if that was a bad idea.

File: docs/src/keyboard_configuration.md [hunk 1]
```diff
@@ -151,9 +151,12 @@ The key string should follow several rules:
     7. Use `"TT(n)"` to create a layer activate or tap toggle action, `n` is the layer number
     8. Use `"TG(n)"` to create a layer toggle action, `n` is the layer number
     9. Use `"TO(n)"` to create a layer toggle only action (activate layer `n` and deactivate all other layers), `n` is the layer number
-    
+
   The definitions of those operations are same with QMK, you can found [here](https://docs.qmk.fm/#/feature_layers). If you want other actions, please [fire an issue](https://github.com/HaoboGu/rmk/issues/new).
 
+4. For modifier-tap-hold, use `MTH(key, modifier)` where the modifier can be a chain like explained on point 1. For example for a Home row modifier config you can use `MTH(F,LShift)`
```
Comment (HaoboGu) [inline docs/src/keyboard_configuration.md]: I prefer `MT`, which is same with QMK, and consistent with `LT`, what do you think?

File: rmk-macro/src/layout.rs [hunk 2]
```diff
@@ -37,6 +37,43 @@ fn expand_row(row: Vec<String>) -> TokenStream2 {
     quote! { [#(#keys), *] }
 }
 
+/// Get modifier combination, in types of mod1 | mod2 | ...
```
Comment (HaoboGu) [inline rmk-macro/src/layout.rs:76]: Please add documentation here to specify what the five `bool`s represent. And there's a minor format issue: 
`){` -> `) {`

**Comment by HaoboGu** [comment]
> I see that Via does not have a binding for the TapHold.

Yes, via/vial supports only ModifierTapHold and LayerTapHold. Only basic keycodes are supported in normal TapHold, and they're not exposed right now.

File: docs/src/keyboard_configuration.md (hunk 1 above)
Comment (hariseldon78) [reply docs/src/keyboard_configuration.md]: sure, i didn't know about the QMK MT macro. I'll keep TH for the other one.

File: rmk-macro/src/layout.rs (hunk 2 above)
Comment (hariseldon78) [reply rmk-macro/src/layout.rs:76]: I decided to use a struct and refactor away the tuples. Also i implemented the macro expansion in the struct impl. Let me know if I did something wrong, I'm not expert on macros. The code compile, I'll test it now on the hardware.

Comment (hariseldon78) [reply rmk-macro/src/layout.rs:76]: i confirm the hardware test is working for MT(A,LShift) .

**Review by HaoboGu — APPROVED** [review]
Thanks very much!

**Pull request merged**